ktor-server/ktor-server-plugins/ktor-server-sessions/common/src/io/ktor/server/sessions/SessionsBuilder.kt

```diff
@@ -316,11 +316,14 @@ internal constructor(
     public val type: KClass<S>,
     public val typeInfo: KType
 ) {
+    private var _serializer: SessionSerializer<S>? = null
 
     /**
      * Specifies a serializer used to serialize session data.
      */
-    public var serializer: SessionSerializer<S> = defaultSessionSerializer(typeInfo)
+    public var serializer: SessionSerializer<S>
+        set(value) { _serializer = value }
+        get() = _serializer ?: defaultSessionSerializer(typeInfo)
```

Comment:
Won't it be a problem that default session serialized could be created several times?

Reply:
Yeah, that's a good point.  I assumed from the name "Builder" that it would only reference the fields a single time for populating the the static instance, but it appears to not be the case.